pkg/newstorage/newstorage.go

```diff
@@ -8,6 +8,12 @@ package newstorage
 
 import "errors"
 
+// BulkOperationError represents one or more errors that occurred during a bulk operation.
+type BulkOperationError interface {
+	error
+	Errors() map[int]error // Errors returns a mapping of failed operation indices to their associated errors.
```

Comment:
why a map[int] instead of a slice?

Reply:
My thinking is that a caller might want the ability to know exactly which operation failed.

Reply:
Changed to slice.